static/app/components/tabs/tab.tsx

```diff
@@ -0,0 +1,181 @@
+import {useEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import {useTab} from '@react-aria/tabs';
+import {mergeProps} from '@react-aria/utils';
+import {TabListState} from '@react-stately/tabs';
+import {Node, Orientation} from '@react-types/shared';
+
+import space from 'sentry/styles/space';
+
+interface TabProps<T> {
+  item: Node<T>;
+  orientation: Orientation;
+  overflowing: boolean;
+  setRef: (ref: React.RefObject<HTMLLIElement>) => void;
+  state: TabListState<T>;
+}
+
+/**
+ * Renders a single tab item. This should not be imported directly into any
+ * page/view – it's only meant to be used by <TabsList />. See the correct
+ * usage in tabs.stories.js
+ */
+export function Tab<T>({item, state, orientation, overflowing, setRef}: TabProps<T>) {
+  const ref = useRef<HTMLLIElement>(null);
+
+  const {key, rendered} = item;
+  const {tabProps, isSelected, isDisabled} = useTab({key}, state, ref);
+
+  useEffect(() => {
```

Comment:
Do you need `setRef`? Couldn't you do a `forwardRef` here and pass `ref={element => ...}` in `TabList`?

Reply:
Just tried this and got a type error from `useTab`. It expects a non-null ref object.

<img width="1040" alt="Screen Shot 2022-09-15 at 3 46 40 PM" src="https://user-images.githubusercontent.com/44172267/190521675-b2fbe097-6a8e-4bd4-b613-47fd44230015.png">

Reply:
Hmm look like you might be able to use `useObjectRef` from `@react-aria/utils` to convert the forwarded ref to an object ref: https://github.com/adobe/react-spectrum/pull/2293

Reply:
Ooo nice it worked! https://github.com/getsentry/sentry/pull/38886/commits/55629662b1e7b9f174ca215e5b9b46725a230c0f